### PR TITLE
feat: Performance improvement for query + UI use post-FC upload endpoint + add FC featuretoggle

### PR DESCRIPTION
**Pull request by rikdepeuter**

### CODE REVIEW

File: paket.dependencies [hunk 1]
```diff
@@ -65,7 +65,7 @@ nuget Serilog.Settings.Configuration 3.3.0
 nuget Serilog.Sinks.Seq 5.1.1
 nuget Swashbuckle.AspNetCore.SwaggerUI 6.2.3
 
-nuget FeatureToggle.Net6 1.0.1
+nuget FeatureToggle.Common 4.0.2
```
Comment (lvermeulen) [inline paket.dependencies:68]: why change from .Net6?

File: paket.lock [hunk 2]
```diff
@@ -469,10 +469,8 @@ NUGET
       Fake.Core.Trace (>= 5.23)
       FSharp.Core (>= 6.0)
     Fare (2.2.1)
-    FeatureToggle.Common.Net6 (1.0.1)
-    FeatureToggle.Net6 (1.0.1)
-      FeatureToggle.Common.Net6 (>= 1.0.1)
-      Microsoft.Extensions.Configuration.Json (>= 6.0)
+    FeatureToggle.Common (4.0.2)
```
Comment (lvermeulen) [inline paket.lock:472]: why change from .Net6?

File: paket.lock [hunk 3]
```diff
@@ -469,10 +469,8 @@ NUGET
...
+      NETStandard.Library (>= 1.6.1)
```
Comment (lvermeulen) [inline paket.lock:473]: why change from .Net6?

File: src/RoadRegistry.BackOffice.ExtractHost/RoadRegistry.BackOffice.ExtractHost.csproj [hunk 4]
```diff
@@ -24,7 +24,10 @@
     <Content Include="Dockerfile" CopyToOutputDirectory="Always" />
     <Content Include="init.sh" CopyToOutputDirectory="Always" />
     <Content Include="appsettings.json" CopyToOutputDirectory="Always" />
-    <Content Include="appsettings.*.json" CopyToOutputDirectory="Always" />
+    <Content Include="appsettings.*.json">
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.ExtractHost/RoadRegistry.BackOffice.ExtractHost.csproj:27]: remove this block (i.e. restore to what it was)

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/GradeSeparatedJunctionArchiveWriter.cs [hunk 5]
```diff
@@ -28,10 +29,8 @@ public GradeSeparatedJunctionArchiveWriter(RecyclableMemoryStreamManager manager
         if (request == null) throw new ArgumentNullException(nameof(request));
         if (context == null) throw new ArgumentNullException(nameof(context));
 
-        var junctions =
-            await context.GradeSeparatedJunctions
-                .InsideContour(request.Contour)
-                .ToListAsync(cancellationToken);
+        var junctions = await context.GradeSeparatedJunctions.ToListWithPolygonials(request.Contour, (dbSet, polygon) => dbSet.InsideContour(polygon), x => x.Id, cancellationToken);
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/GradeSeparatedJunctionArchiveWriter.cs:32]: use a line-separated method chain (like what it was)

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/IntegrationRoadNodesToZipArchiveWriter.cs [hunk 6]
```diff
@@ -32,21 +33,16 @@ public IntegrationRoadNodesToZipArchiveWriter(RecyclableMemoryStreamManager mana
 
         const int integrationBufferInMeters = 350;
 
-        var nodesInContour = await context.RoadNodes
-            .InsideContour(request.Contour)
-            .ToListAsync(cancellationToken);
+        var nodesInContour = await context.RoadNodes.ToListWithPolygonials(request.Contour, (dbSet, polygon) => dbSet.InsideContour(polygon), x => x.Id, cancellationToken);
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/IntegrationRoadNodesToZipArchiveWriter.cs:36]: use a line-separated method chain (like what it was)

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/IntegrationRoadNodesToZipArchiveWriter.cs [hunk 7]
```diff
@@ -32,21 +33,16 @@ public IntegrationRoadNodesToZipArchiveWriter(RecyclableMemoryStreamManager mana
...
         var geometryForNodesInContour = GeometryConfiguration.GeometryFactory
             .BuildGeometry(nodesInContour.Select(node => node.Geometry));
 
         var boundaryForNodes = geometryForNodesInContour.ConvexHull();
         var boundaryWithIntegrationBuffer = boundaryForNodes.Buffer(integrationBufferInMeters);
 
-        var nodesInIntegrationBuffer = await context.RoadNodes
-            .InsideContour(boundaryWithIntegrationBuffer as IPolygonal)
-            .ToListAsync(cancellationToken);
+        var nodesInIntegrationBuffer = await context.RoadNodes.ToListWithPolygonials(boundaryWithIntegrationBuffer as IPolygonal, (dbSet, polygon) => dbSet.InsideContour(polygon), x => x.Id, cancellationToken);
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/IntegrationRoadNodesToZipArchiveWriter.cs:43]: use a line-separated method chain (like what it was)

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/IntegrationRoadSegmentsToZipArchiveWriter.cs [hunk 8]
```diff
@@ -41,18 +42,14 @@ public class IntegrationRoadSegmentsToZipArchiveWriter : IZipArchiveWriter<Edito
 
         const int integrationBufferInMeters = 350;
 
-        var segmentsInContour = await context.RoadSegments
-            .InsideContour(request.Contour)
-            .ToListAsync(cancellationToken);
+        var segmentsInContour = await context.RoadSegments.ToListWithPolygonials(request.Contour, (dbSet, polygon) => dbSet.InsideContour(polygon), x => x.Id, cancellationToken);
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/IntegrationRoadSegmentsToZipArchiveWriter.cs:45]: use a line-separated method chain (like what it was)

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/IntegrationRoadSegmentsToZipArchiveWriter.cs [hunk 9]
```diff
@@ -41,18 +42,14 @@ public class IntegrationRoadSegmentsToZipArchiveWriter : IZipArchiveWriter<Edito
...
         var geometryForSegmentsInContour = GeometryConfiguration.GeometryFactory
             .BuildGeometry(segmentsInContour.Select(segment => segment.Geometry));
 
         var boundaryForSegments = geometryForSegmentsInContour.ConvexHull();
         var boundaryWithIntegrationBuffer = boundaryForSegments.Buffer(integrationBufferInMeters);
 
-        var segmentsInIntegrationBuffer = await context.RoadSegments
-            .InsideContour(boundaryWithIntegrationBuffer as IPolygonal)
-            .ToListAsync(cancellationToken);
+        var segmentsInIntegrationBuffer = await context.RoadSegments.ToListWithPolygonials(boundaryWithIntegrationBuffer as IPolygonal, (dbSet, polygon) => dbSet.InsideContour(polygon), x => x.Id, cancellationToken);
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/IntegrationRoadSegmentsToZipArchiveWriter.cs:52]: use a line-separated method chain (like what it was)

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadNodesToZipArchiveWriter.cs [hunk 10]
```diff
@@ -28,8 +30,7 @@ public RoadNodesToZipArchiveWriter(RecyclableMemoryStreamManager manager, Encodi
         if (request == null) throw new ArgumentNullException(nameof(request));
         if (context == null) throw new ArgumentNullException(nameof(context));
 
-        var nodes =
-            await context.RoadNodes.InsideContour(request.Contour).ToListAsync(cancellationToken);
+        var nodes = await context.RoadNodes.ToListWithPolygonials(request.Contour, (dbSet, polygon) => dbSet.InsideContour(polygon), x => x.Id, cancellationToken);
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadNodesToZipArchiveWriter.cs:33]: use a line-separated method chain (like what it was)

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentLaneAttributesToZipArchiveWriter.cs [hunk 11]
```diff
@@ -28,9 +29,8 @@ public RoadSegmentLaneAttributesToZipArchiveWriter(RecyclableMemoryStreamManager
         if (request == null) throw new ArgumentNullException(nameof(request));
         if (context == null) throw new ArgumentNullException(nameof(context));
 
-        var attributes = await context.RoadSegmentLaneAttributes
-            .InsideContour(request.Contour)
-            .ToListAsync(cancellationToken);
+        var attributes = await context.RoadSegmentLaneAttributes.ToListWithPolygonials(request.Contour, (dbSet, polygon) => dbSet.InsideContour(polygon), x => x.Id, cancellationToken);
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentLaneAttributesToZipArchiveWriter.cs:32]: use a line-separated method chain (like what it was)

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentNationalRoadAttributesToZipArchiveWriter.cs [hunk 12]
```diff
@@ -29,9 +30,7 @@ public class RoadSegmentNationalRoadAttributesToZipArchiveWriter : IZipArchiveWr
         if (request == null) throw new ArgumentNullException(nameof(request));
         if (context == null) throw new ArgumentNullException(nameof(context));
 
-        var attributes = await context.RoadSegmentNationalRoadAttributes
-            .InsideContour(request.Contour)
-            .ToListAsync(cancellationToken);
+        var attributes = await context.RoadSegmentNationalRoadAttributes.ToListWithPolygonials(request.Contour, (dbSet, polygon) => dbSet.InsideContour(polygon), x => x.Id, cancellationToken);
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentNationalRoadAttributesToZipArchiveWriter.cs:33]: use a line-separated method chain (like what it was)

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentNumberedRoadAttributesToZipArchiveWriter.cs [hunk 13]
```diff
@@ -29,10 +30,7 @@ public class RoadSegmentNumberedRoadAttributesToZipArchiveWriter : IZipArchiveWr
         if (request == null) throw new ArgumentNullException(nameof(request));
         if (context == null) throw new ArgumentNullException(nameof(context));
 
-        var attributes = await context.RoadSegmentNumberedRoadAttributes
-            .InsideContour(request.Contour)
-            .ToListAsync(cancellationToken);
-        var dbfEntry = archive.CreateEntry("eAttGenumweg.dbf");
+        var attributes = await context.RoadSegmentNumberedRoadAttributes.ToListWithPolygonials(request.Contour, (dbSet, polygon) => dbSet.InsideContour(polygon), x => x.Id, cancellationToken); var dbfEntry = archive.CreateEntry("eAttGenumweg.dbf");
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentNumberedRoadAttributesToZipArchiveWriter.cs:33]: use a line-separated method chain (like what it was)

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentSurfaceAttributesToZipArchiveWriter.cs [hunk 14]
```diff
@@ -28,9 +29,8 @@ public RoadSegmentSurfaceAttributesToZipArchiveWriter(RecyclableMemoryStreamMana
         if (request == null) throw new ArgumentNullException(nameof(request));
         if (context == null) throw new ArgumentNullException(nameof(context));
 
-        var attributes = await context.RoadSegmentSurfaceAttributes
-            .InsideContour(request.Contour)
-            .ToListAsync(cancellationToken);
+        var attributes = await context.RoadSegmentSurfaceAttributes.ToListWithPolygonials(request.Contour, (dbSet, polygon) => dbSet.InsideContour(polygon), x => x.Id, cancellationToken);
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentSurfaceAttributesToZipArchiveWriter.cs:32]: use a line-separated method chain (like what it was)

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentWidthAttributesToZipArchiveWriter.cs [hunk 15]
```diff
@@ -28,9 +29,8 @@ public RoadSegmentWidthAttributesToZipArchiveWriter(RecyclableMemoryStreamManage
         if (request == null) throw new ArgumentNullException(nameof(request));
         if (context == null) throw new ArgumentNullException(nameof(context));
 
-        var attributes = await context.RoadSegmentWidthAttributes
-            .InsideContour(request.Contour)
-            .ToListAsync(cancellationToken);
+        var attributes = await context.RoadSegmentWidthAttributes.ToListWithPolygonials(request.Contour, (dbSet, polygon) => dbSet.InsideContour(polygon), x => x.Id, cancellationToken); ;
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentWidthAttributesToZipArchiveWriter.cs:32]: use a line-separated method chain (like what it was)

File: src/RoadRegistry.BackOffice/Extensions/NetTopologySuiteExtensions.cs [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Threading;
+
+public static class NetTopologySuiteExtensions
+{
+    public static IPolygonal[] GetPolygonals(this IPolygonal polygonal)
+    {
+        if (polygonal is Polygon polygon) return new IPolygonal[] { polygon };
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice/Extensions/NetTopologySuiteExtensions.cs:13]: use braces on all if statements

File: test/RoadRegistry.Tests/TestStartup.cs [hunk 17]
```diff
@@ -39,6 +40,11 @@ public void ConfigureHost(IHostBuilder hostBuilder)
         var availableModuleAssemblyCollection = DetermineAvailableAssemblyCollection();
 
         hostBuilder
+            .ConfigureAppConfiguration((hostContext, configurationBuilder) =>
+            {
+                configurationBuilder
+                    .AddJsonFile("appsettings.test.json", optional: true, reloadOnChange: true);
```
Comment (lvermeulen) [inline test/RoadRegistry.Tests/TestStartup.cs:46]: why not simply use appsettings.json? it's a test.

File: paket.dependencies (hunk 1 above)
Comment (rikdepeuter) [reply paket.dependencies:68]: When using the `IFeatureToggle` interface in the Backoffice API there became a conflict of 2 assemblies containing it. The FeatureToggle.Common.Net6 (v1.0.0.0) and FeatureToggle.Common (v4.0.2.0).
I then checked streetname-registry and saw that their `IFeatureToggle` came from the FeatureToggle.Common (v4.0.2.0) assembly.

Now upon further investigation I see that actually the FeatureToggle.Common assembly comes from the nuget `Be.Vlaanderen.Basisregisters.DataDog.Tracing.Autofac` even though it doesn't have the FeatureToggle.Common listed as an dependency.
`.nuget\packages\be.vlaanderen.basisregisters.datadog.tracing.autofac\5.0.0\lib\net6.0\FeatureToggle.Common.dll`

File: src/RoadRegistry.BackOffice.ExtractHost/RoadRegistry.BackOffice.ExtractHost.csproj (hunk 4 above)
Comment (rikdepeuter) [reply src/RoadRegistry.BackOffice.ExtractHost/RoadRegistry.BackOffice.ExtractHost.csproj]: Done

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/GradeSeparatedJunctionArchiveWriter.cs (hunk 5 above)
Comment (rikdepeuter) [reply src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/GradeSeparatedJunctionArchiveWriter.cs]: Done

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/IntegrationRoadNodesToZipArchiveWriter.cs (hunk 6 above)
Comment (rikdepeuter) [reply src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/IntegrationRoadNodesToZipArchiveWriter.cs]: Done

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/IntegrationRoadNodesToZipArchiveWriter.cs (hunk 7 above)
Comment (rikdepeuter) [reply src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/IntegrationRoadNodesToZipArchiveWriter.cs]: Done

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/IntegrationRoadSegmentsToZipArchiveWriter.cs (hunk 9 above)
Comment (rikdepeuter) [reply src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/IntegrationRoadSegmentsToZipArchiveWriter.cs]: Done

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadNodesToZipArchiveWriter.cs (hunk 10 above)
Comment (rikdepeuter) [reply src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadNodesToZipArchiveWriter.cs]: Done

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentEuropeanRoadAttributesToZipArchiveWriter.cs [hunk 18]
```diff
@@ -29,9 +30,7 @@ public class RoadSegmentEuropeanRoadAttributesToZipArchiveWriter : IZipArchiveWr
         if (request == null) throw new ArgumentNullException(nameof(request));
         if (context == null) throw new ArgumentNullException(nameof(context));
 
-        var attributes = await context.RoadSegmentEuropeanRoadAttributes
-            .InsideContour(request.Contour)
-            .ToListAsync(cancellationToken);
+        var attributes = await context.RoadSegmentEuropeanRoadAttributes.ToListWithPolygonials(request.Contour, (dbSet, polygon) => dbSet.InsideContour(polygon), x => x.Id, cancellationToken);
```
[the remark this answers is not in the document]
Comment (rikdepeuter) [reply src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentEuropeanRoadAttributesToZipArchiveWriter.cs]: Done

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentLaneAttributesToZipArchiveWriter.cs (hunk 11 above)
Comment (rikdepeuter) [reply src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentLaneAttributesToZipArchiveWriter.cs]: Done

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentNationalRoadAttributesToZipArchiveWriter.cs (hunk 12 above)
Comment (rikdepeuter) [reply src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentNationalRoadAttributesToZipArchiveWriter.cs]: Done

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentNumberedRoadAttributesToZipArchiveWriter.cs (hunk 13 above)
Comment (rikdepeuter) [reply src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentNumberedRoadAttributesToZipArchiveWriter.cs]: Done

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentSurfaceAttributesToZipArchiveWriter.cs (hunk 14 above)
Comment (rikdepeuter) [reply src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentSurfaceAttributesToZipArchiveWriter.cs]: Done

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentWidthAttributesToZipArchiveWriter.cs (hunk 15 above)
Comment (rikdepeuter) [reply src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentWidthAttributesToZipArchiveWriter.cs]: Done

File: src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentsToZipArchiveWriter.cs [hunk 19]
```diff
@@ -37,9 +38,8 @@ public class RoadSegmentsToZipArchiveWriter : IZipArchiveWriter<EditorContext>
         if (request == null) throw new ArgumentNullException(nameof(request));
         if (context == null) throw new ArgumentNullException(nameof(context));
 
-        var segments = await context.RoadSegments
-            .InsideContour(request.Contour)
-            .ToListAsync(cancellationToken);
+        var segments = await context.RoadSegments.ToListWithPolygonials(request.Contour, (dbSet, polygon) => dbSet.InsideContour(polygon), x => x.Id, cancellationToken);
```
[the remark this answers is not in the document]
Comment (rikdepeuter) [reply src/RoadRegistry.BackOffice.ZipArchiveWriters/ExtractHost/RoadSegmentsToZipArchiveWriter.cs]: Done

File: src/RoadRegistry.BackOffice/Extensions/NetTopologySuiteExtensions.cs (hunk 16 above)
Comment (rikdepeuter) [reply src/RoadRegistry.BackOffice/Extensions/NetTopologySuiteExtensions.cs]: Done

File: test/RoadRegistry.Tests/TestStartup.cs (hunk 17 above)
Comment (rikdepeuter) [reply test/RoadRegistry.Tests/TestStartup.cs]: Done

File: paket.dependencies (hunk 1 above)
Comment (lvermeulen) [reply paket.dependencies:68]: so why not use .Net6 ?

**Pull request merged**